jbmc/src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -307,7 +307,7 @@ java_string_library_preprocesst::convert_exprt_to_string_exprt(
 {
   PRECONDITION(implements_java_char_sequence_pointer(expr_to_process.type()));
   const refined_string_exprt string_expr =
-    decl_string_expr(loc, symbol_table, init_code);
+    decl_string_expr(loc, loc.get_function(), symbol_table, init_code);
```

Comment:
This is most likely not the correct name, you'd want the name that is known to the symbol table. Really really really only use `source_locationt::get_function` if you are generating output to humans.